android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/IngestionMetadataErichmentPlugin.kt

```diff
@@ -0,0 +1,25 @@
+package com.amplitude.ampli
+
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.platform.Plugin
+
+class IngestionMetadataErichmentPlugin : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Enrichment
+    override lateinit var amplitude: com.amplitude.core.Amplitude
+
+    override fun execute(event: BaseEvent): BaseEvent? {
+        if (event.extra == null) event.extra = mapOf()
```

Comment:
nit: single line if's can lead to trouble please add braces

Reply:
Can probably make the following logic more concise with this

```
val amplitudeExtra = event.extra ? event.extra["amplitude"] ?: mapOf() : mapOf();

amplitudeExta.plus(this.ingestionMetadata.toJson())

event.extra["amplitude"] = amplitudeExtra;
```

Reply:
Is `event.extra["amplitude"]` a right syntax for kotlin?
Also since we didn't have any type defined fo event.extra.amplitude.  We need it to cover to the right format in order to do amplitudeExtra.plus(xxx)

Also, amplitude might have different types of info later. do we need save `ingestionMetadata` in amplitude.ingestionMetadata key?

Hope this is better. 

```
        val amplitudeExtra = (event.extra?.get("amplitude") ?: mapOf<String, Any>()) as Map<String, Any>
        val ingestionMetadataMap = mapOf(
            "ingestionMetadata" to  ingestionMetadata.toJsonObject()
        )
        amplitudeExtra.plus(ingestionMetadataMap)
        event.extra = event.extra?.plus(mapOf("amplitude" to amplitudeExtra))
        return event
```

Reply:
Thanks, that was pseudo code above. Didn't have the fortitude to type out the full kotlin.